src/modules/0.0.21/aws_security_group/entity/index.ts

```diff
@@ -97,4 +106,13 @@ export class SecurityGroupRule {
     nullable: true,
   })
   description?: string;
+
+  @Check(
+    'Check_security_or_ip_permissions',
+    `("source_security_group" IS NULL AND ("from_port" IS NOT NULL OR "to_port" IS NOT NULL OR "cidr_ipv4" IS NOT NULL OR "cidr_ipv6" IS NOT NULL)) OR ("source_security_group" IS NOT NULL AND ("from_port" IS NULL AND "to_port" IS NULL AND "cidr_ipv4" IS NULL AND "cidr_ipv6" IS NULL))`,
+  )
+  @Column({
+    nullable: true,
+  })
+  sourceSecurityGroup?: string;
```

Comment:
Why is this not a `SecurityGroup` entity here?

Reply:
That's the first option i tried.. but check https://github.com/iasql/iasql-engine/actions/runs/3125493049/jobs/5069921967#step:7:16399 
For some reason, it is not populating the SourceSecurityGroup (it doesn't even show the field as undefined). I was wondering if that could be caused by having 2 fks to the same table... so after quite a lot of tries, i ended just passing the name as string. Do you know what can be causing that?

Reply:
when you tried was the `eager` option set to `true`?
```
@Column({
    nullable: true,
    eager: true,
  })
  sourceSecurityGroup?: SecurityGroup;
```

Reply:
i updated to use eager:true and then it brings the entity. So far i am not facing any problems with recursion